Style content container of ISAD index page

replaceafill opened the pull request:
This:

* adds a `content` slot to the ISAD index template and styles all its partials and components to use BS5 classes and a 3-9 column layout
* modifies the existing `render_show` helper to render fields and values using BS5 classes and introduces a new `isSubField` option to render fields inside of fields using `flex` containers
* attaches the `expander.js` behaviour to top level field values

jraddaoui commented on plugins/arDominionB5Plugin/modules/arStorageService/templates/_aipDownload.php:
```diff
@@ -0,0 +1,25 @@
+<div class="field <?php echo render_b5_show_field_css_classes(); ?>">
+    <?php echo render_b5_show_label(__('File UUID'), ['isSubField' => true]); ?>
+    <div class="aip-download <?php echo render_b5_show_value_css_classes(['isSubField' => true]); ?>">
+        <?php echo render_value_inline($resource->object->objectUUID); ?>
+        <?php if ($sf_user->checkModuleActionAccess('arStorageService', 'extractFile')) { ?>
+          <a href="<?php echo url_for([$resource, 'module' => 'arStorageService', 'action' => 'extractFile']); ?>" target="_blank">
+            <i class="fa fa-download"></i>
+            <?php echo __('Download file'); ?>
+          </a>
+        <?php } ?>
+    </div>
```
Indentation.

jraddaoui commented on plugins/arDominionB5Plugin/modules/arStorageService/templates/_aipDownload.php:
```diff
@@ -0,0 +1,25 @@
+        <?php echo render_value_inline($resource->object->objectUUID); ?>
+        <?php if ($sf_user->checkModuleActionAccess('arStorageService', 'extractFile')) { ?>
+          <a href="<?php echo url_for([$resource, 'module' => 'arStorageService', 'action' => 'extractFile']); ?>" target="_blank">
+            <i class="fa fa-download"></i>
```
Please, add `aria-hidden="true"` attribute.

jraddaoui commented on plugins/arDominionB5Plugin/modules/arStorageService/templates/_aipDownload.php:
```diff
@@ -0,0 +1,25 @@
+            <?php echo __('Download AIP'); ?>
+          </a>
+        <?php } ?>
+    </div>
```
Same two things.

jraddaoui commented on plugins/arDominionB5Plugin/modules/digitalobject/templates/_metadata.php:
```diff
@@ -20,90 +20,94 @@
-                </div>
+                <div class="<?php echo render_b5_show_field_css_classes(); ?>">
+
+                  <h3 class="<?php echo render_b5_show_label_css_classes(); ?>"><?php echo __('Original file'); ?> <i class="fa fa-archive text-dark<?php if (!$canAccessOriginalFile) { ?> inactive text-muted<?php } ?>" aria-hidden="true"></i></h3>
```
I have not checked how it looks, but you may want to remove the space between the heading text and the icon and use `ms-1` (or 2) class in the icon.

jraddaoui replied:
And other icons below.

jraddaoui commented on plugins/arDominionB5Plugin/modules/digitalobject/templates/_metadata.php:
```diff
@@ -122,130 +126,136 @@
-                  <?php } ?>
+                    <?php if ($showMasterFileGeolocation) { ?>
+                      <?php echo render_show(__('Latitude'), render_value_inline($latitude), ['fieldLabel' => 'latitude', 'isSubField' => true]); ?>
+                      <?php echo render_show(__('Longitude'), render_value_inline($longitude), ['fieldLabel' => 'longitude', 'isSubField' => true]); ?>
```
Nice!

jraddaoui commented on lib/helper/QubitHelper.php:
```diff
@@ -174,6 +178,101 @@ function render_show($label, $value, $options = [])
+
+    return <<<contents
+<{$tag}{$cssClass}>{$content}</{$tag}>
+contents;
```
Could this be `return "<{$tag}{$cssClass}>{$content}</{$tag}>";` directly? I hate heredocs.

jraddaoui reviewed (approved):
Great work @replaceafill!

You took one of the big ones so early and made it work without custom styles. As we chatted, I found a bit confusing to have all those helper functions but I see some of them are used from the outside and it should allow easier updates in the future.

I added a few comments below, nothing big. Thanks!!

replaceafill committed:
Style content container of ISAD index page

* adds a `content` slot to the ISAD index template and styles all its
  partials and components to use BS5 classes and a 3-9 column layout
* modifies the existing `render_show` helper to render fields and
  values using BS5 classes and introduces a new `isSubField` option to
  render fields inside of fields using `flex` containers
* attaches the `expander.js` behaviour to top level field values
